tasks/dataset-split/main.py

```diff
@@ -0,0 +1,129 @@
+        splitDataset.saveClasses(datasetClasses)
+
+        for sample in sampleChunk:
+            sample.unzip()
+            addedSample = splitDataset.add(sample.imagePath)
+            
+            tmpAnotation = sample.load().annotation
+            if tmpAnotation is not None:
+                addedSample.saveAnnotation(tmpAnotation)
+
+            try:
+                tmpMetadata = sample.loadMetadata()
+                addedSample.saveMetadata(tmpMetadata)
+            except FileNotFoundError:
+                logging.warning("Metadata file not found")
+            except ValueError:
+                logging.warning("Invalid metadata type")
+        
+        splitDatasetsList.append(splitDataset)
+
+    return splitDatasetsList
+
+
+def customDatasetSplit(splitSamples: list[list[CustomSample]], projectID: int) -> list[CustomDataset]:  
+    splitDatasetsList: list[CustomDataset] = []
+
+    for index, sampleChunk in enumerate(splitSamples):
+        splitDataset = CustomDataset.createDataset(f"{currentTaskRun().id}-newDataset-{index}", projectID)
+
+        for sample in sampleChunk:
+            sample.unzip()
+            splitDataset.add(sample.zipPath)
+
+        splitDatasetsList.append(splitDataset)
+
+    return splitDatasetsList
+
+
+def sequenceDatasetSplit(splitSamples: list[list[SequenceSample]], metadata: Any, projectID: int) -> list[CustomDataset]:
+    metadataAddress = list(metadata.load().folderContent)[0] #address where the file metadata is located in the form of a string
+    
+    with open(metadataAddress, mode="r", newline="") as file:
+        reader = csv.DictReader(file)
+        originalMetadata: list[dict] = []
+        
+        for row in reader:
+            originalMetadata.append(dict(row))
+
+    splitDatasetsList: list[CustomDataset] = []
+    
+    for index, sampleChunk in enumerate(splitSamples):
+        splitDataset = CustomDataset.createDataset(f"{currentTaskRun().id}-newDataset-{index}", projectID)
+        splitMetadataList: list[dict] = []
+        
+        for sample in sampleChunk:
+            sample.unzip()
+            splitDataset.add(sample.zipPath)
+            
+            fieldNames = list(originalMetadata[0].keys())
+            
+            for oneMetadata in originalMetadata:
+                if sample.name.startswith(oneMetadata[fieldNames[0]].split("_")[0]):
+                    splitMetadataList.append(oneMetadata)
+
+        metadataCSV = f"_metadata_{index}.csv"
+        with open(metadataCSV, "w", newline="") as file:
+            writer = csv.DictWriter(file, fieldnames=fieldNames)
+            writer.writeheader()
+            writer.writerows(splitMetadataList)
+
+        metadataZIP = f"_metadata_{index}.zip"
+        with zipfile.ZipFile(metadataZIP, "w") as zipFile:
+            zipFile.write(metadataCSV)
+
+        splitDataset.add(metadataZIP)
+
+    splitDatasetsList.append(splitDataset)
+          
+    return splitDatasetsList
+
+
+def main() -> None:
```

Comment:
Let's add some logs to this template which display info about datasets as they are being created as well as samples, and also information about how many samples will each of the datasets contain after split, etc... Whatever info is going to be relevant to the user.